### ISSUE TITLE
Incorrect page overlap display when moving within one book page

### ISSUE THREAD

**Issue by jonnyl2** [koreader/koreader#10652]
* KOReader version: 2023.05.1-8
* Device: Kobo Aura H2O

#### Issue

Page overlap doesn't work correctly when moving backward within a book page.

#### Steps to reproduce

Have fixed-layout document open and view set to Page Crop: auto; Fit width, Zoom to content, View mode page; Reflow off.

Page overlap is activated and set to Gray out.

1 Book page will be split vertically into 2 screens. Go forward one screen page and then back one screen page.

**1st screen of page 1:**
(correct)

![Reader_2023-07-06_183700](https://github.com/koreader/koreader/assets/95502269/1cb9c554-00da-45e7-9832-af2d5356dc3a)

**2nd screen of page 1:**
(correct shading)

![Reader_2023-07-06_183710](https://github.com/koreader/koreader/assets/95502269/136edcf3-3da2-45f2-8037-36595bf05000)

**Going back to previous screen:**
(there should be no gray shading on this screen)

![Reader_2023-07-06_183724](https://github.com/koreader/koreader/assets/95502269/06cade9a-a43e-4042-8f0d-4b4de0cb3bc4)

(When turning back crosses a book page boundary, the shading displays correctly.)

**Comment by NiLuJe** [comment]
Yeah, there are a number of edge-cases where it's subtly (or not so subtly) broken. There's seven billion layers of indirection in this thing, making it a nightmare to deal with.

I don't particularly care about it, so I gave up on that mess after barely being able to fix the more egregious issues with it.

**Comment by jonnyl2** [comment]
OK, that's fair. It's not particularly urgent.

**Comment by NiLuJe** [comment]
It's a nice feature, and I wish it wasn't such a pain to deal with every time there's an issue with it ;).

**Comment by mergen3107** [comment]
Yeah, there are multiple issues with page overlap. Especially when you have 4 sector of zoom to go through, and when going from one vertical to another it draws only horizontal line :D (I use the line instead of arrows/shades)
